### PR TITLE
Add project form

### PR DESCRIPTION
**Pull request by queq1890**
## Story URL
https://trello.com/c/vZ32ZMYJ

## What I did
- Create a new project form 
  - Add `react-select` package to handle multiple members selection
  - Add `PeojectPage`
  - Add `ProjectFormCreate` component

- Fix Sidebar roles
  - use `rolesArray` instead of `roles` ( because database structure will be changed in the serverside )

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "react-loadable": "^5.5.0",
     "react-router": "^4.3.1",
     "react-router-dom": "^4.3.1",
+    "react-select": "^2.1.1",
```
Comment (queq1890) [inline package.json:74]: I used `react-select` to pick multiple members
https://react-select.com/home

File: src/index.js [hunk 2]
```diff
@@ -14,22 +14,18 @@ import App from './containers/App';
 import history from './utils/history';
 import 'semantic-ui-css/semantic.min.css';
 import './ress.min.css';
+import './common.style';
```
Comment (queq1890) [inline src/index.js:17]: Just injecting this CSS globally from `'./common.style';`
```
 *:focus {
    outline: none;
  }
```

**Comment by queq1890** [comment]
As I and Thuan talked this morning, 
this pull request will be merged  **after `daily report` related pull requests (https://trello.com/c/zgcc1Dwq) are merged.**

 It's because both pull requests have breaking changes and gonna conflict a lot.

File: src/components/ProjectFormCreate/index.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import ErrorMessage from '../ErrorMessage';
+
+const ProjectSchema = Yup.object().shape({
+  title: Yup.string().required('Title is required'),
+  teamLeader: Yup.string().required('Team leader is required'),
+  members: Yup.array().min(1, 'Pick at least 1 member')
+});
+
+const getOptions = users => {
+  const options = [];
```
Comment (ggianglv) [inline src/components/ProjectFormCreate/index.js]: another way is juse use map
`const options = users.map(user => ( {value: user.id, label: user.name })`

File: src/components/ProjectFormCreate/index.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+              title: '',
+              members: [],
+              teamLeader: ''
+            }}
+            enableReinitialize
+            validationSchema={ProjectSchema}
+            onSubmit={async (values, { setSubmitting, setStatus, setErrors, resetForm }) => {
+              const { title } = values;
+              const teamLeader = values.teamLeader.value;
+              const members = [];
```
Comment (ggianglv) [inline src/components/ProjectFormCreate/index.js]: same as above. use`map`

File: src/components/SelectInput/index.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    borderColor: '#e0b4b4',
+    fontSize: '14px'
+  })
+};
+
+class SelectInput extends React.Component {
+  handleChange = value => {
+    // this is going to call setFieldValue and manually update values.members
+    const { name } = this.props;
+    this.props.onChange(name, value);
```
Comment (ggianglv) [inline src/components/SelectInput/index.js]:
```
const { name, onChange } = this.props;
onChange(name, value);
```

File: src/components/SelectInput/index.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  handleBlur = () => {
+    // this is going to call setFieldTouched and manually update touched.members
+    const { name } = this.props;
```
Comment (ggianglv) [inline src/components/SelectInput/index.js]: const { name, onBlur } = this.props;
onBlur(name, true);

File: src/components/SelectInput/index.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    return (
+      <FormGroup>
+        <FormFieldStyles>
+          <LabelStyles error={error}>{name}</LabelStyles>
+          <SelectStyles
+            placeholder={placeholder}
+            options={options}
+            isMulti={!!isMulti}
+            onChange={this.handleChange}
+            onBlur={() => this.handleBlur}
```
Comment (ggianglv) [inline src/components/SelectInput/index.js]: maybe you miss type here `onBlur={this.handleBlur}`

**Review by phuongthuan — APPROVED** [review]
Approved @queq1890 👍

**Pull request merged**